File: analytic-engine/src/main/java/com/mycompany/models/PlotModel.java
package com.mycompany.models;

import java.util.Objects;

public class PlotModel {
    public String _id;
    public String __v;
    public String jobId;
    public String identifier;
    public String identifierType;
    public String xAxis;
    public String xType;
    public String yAxis;
    public String yType;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PlotModel plotModel = (PlotModel) o;
        return Objects.equals(_id, plotModel._id) &&
                Objects.equals(__v, plotModel.__v) &&
                Objects.equals(jobId, plotModel.jobId) &&
                Objects.equals(identifier, plotModel.identifier) &&
                Objects.equals(identifierType, plotModel.identifierType) &&
                Objects.equals(xAxis, plotModel.xAxis) &&
                Objects.equals(xType, plotModel.xType) &&
                Objects.equals(yAxis, plotModel.yAxis) &&
                Objects.equals(yType, plotModel.yType);
    }
}

File: mongodb-service/src/models/plotModel.ts
import * as mongoose from "mongoose";
import { Model } from "mongoose";

/**
 * Plot type, model mongoose document and schema definition. Models are used by
 * Controllers so they are picked up by the swagger. Documents and schemas are used by 
 * Repositories so model defined Mongodb functions can be used.
 */
type PlotType = IPlot & mongoose.Document;
export interface IPlotModel {
    _id?: string;
    jobId: string;
    identifier: string;
    identifierType: string;
    xAxis: string;
    xType: string;
    yAxis: string;
    yType: string;
}

export interface IPlot extends IPlotModel, mongoose.Document {
    _id: string;
}

export const PlotSchema = new mongoose.Schema({
    jobId: {
        required: true,
        type: String
    },
    identifier: {
        required: true,
        type: String
    },
    identifierType: {
        required: true,
        type: String
    },
    xAxis: {
        required: true,
        type: String
    },
    xType: {
        required: true,
        type: String
    },
    yAxis: {
        required: true,
        type: String
    },
    yType: {
        required: true,
        type: String
    },
});

const Plot: Model<IPlot> = mongoose.model<PlotType>("Plot", PlotSchema);
export default Plot;